test/link_checker.rb

```diff
@@ -47,7 +47,7 @@ def navigate_path(path, hash = @toc)
 
     inner_hash = hash[first]
 
-    return inner_hash if rest.empty?
+    return inner_hash if rest.nil? or rest.empty?
```

Comment:
Before Zeitwerk, Rails was loaded when this test was running and Rails in its endless wisdom adds `empty?` to `NilClass`. With Zeitweirk, no Rails is present here, and so no `NilClass.empty?`. Let's catch that via `nil?` first.